aiida/repository/backend/disk_object_store.py

```diff
@@ -118,3 +120,96 @@ def get_object_hash(self, key: str) -> str:
         if self.container.hash_type != 'sha256':
             return super().get_object_hash(key)
         return key
+
+    def maintain(  # type: ignore # pylint: disable=arguments-differ,too-many-branches
+        self,
+        dry_run: bool = False,
+        live: bool = True,
+        pack_loose: bool = None,
+        do_repack: bool = None,
+        clean_storage: bool = None,
+        do_vacuum: bool = None,
+    ) -> dict:
+        """Performs maintenance operations.
+
+        :param full:
+            flag to perform operations that require to stop using the maintained profile.
+        :param override_pack_loose:
+            override flag for forcing the packing of loose files.
+        :param override_do_repack:
+            override flag for forcing the re-packing of already packed files.
+        :param override_clean_storage:
+            override flag for forcing the cleaning of soft-deleted files from the repository.
+        :param override_do_vacuum:
+            override flag for forcing the vacuuming of the internal database when cleaning the repository.
+        :return:
+            a dictionary with information on the operations performed.
+        """
+        from aiida.backends.control import MAINTAIN_LOGGER
+        DOSTORE_LOGGER = MAINTAIN_LOGGER.getChild('disk_object_store')  # pylint: disable=invalid-name
+
+        if live and (do_repack or clean_storage or do_vacuum):
+            overrides = {'do_repack': do_repack, 'clean_storage': clean_storage, 'do_vacuum': do_vacuum}
+            keys = ', '.join([key for key, override in overrides if override is True])  # type: ignore
+            raise ValueError(f'The following overrides were enabled but cannot be if `live=True`: {keys}')
+
+        pack_loose = True if pack_loose is None else pack_loose
```

Comment:
For this type of conditional there is this standard idiom
```suggestion
        pack_loose = pack_loose or True
```

Reply:
Mmm but if the user passes `False` I want it to respect that and what you propose would overwrite it no?

Reply:
Nope, this construction only hits the `or` if the variable is `None`, not just something falsy

Reply:
Mmmm are you sure? I mean, I don't mean to be skeptic of you but I just tried it and it giving me the behavior I am describing:

```python
In [1]: false_var = False

In [2]: true_var = True

In [3]: none_var = None

In [4]: destination = none_var or 'GOOD'

In [5]: destination
Out[5]: 'GOOD'

In [6]: destination = true_var or 'BAD'

In [7]: destination
Out[7]: True

In [8]: destination = false_var or 'BAD'

In [9]: destination
Out[9]: 'BAD'

In [10]: destination = false_var or True  # to check the exact case in case strings have a different behaviour

In [11]: destination
Out[11]: True
```

Would it make a difference if the variables are arguments of the encompassing function? I can't imagine it would...

Reply:
Yep, you are absolutely right. So forget about this comment